adam-core/src/main/scala/org/bdgenomics/adam/rdd/RegionRDDFunctions.scala

```diff
@@ -0,0 +1,187 @@
+/**
+ * Licensed to Big Data Genomics (BDG) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The BDG licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bdgenomics.adam.rdd
+
+import org.apache.spark.SparkContext._
+import org.apache.spark.rdd.RDD
+import org.bdgenomics.adam.models.{ SequenceDictionary, ReferenceRegionWithOrientation, ReferenceRegion }
+import org.bdgenomics.adam.models.ReferenceRegionContext._
+import org.bdgenomics.adam.rich.ReferenceMappingContext.ReferenceRegionReferenceMapping
+import scala.reflect.ClassTag
+import scala.math.max
+
+/**
+ * Functions (joins, filters, groups, counts) that are often invoked on RDD[R] where R is a
+ * type which extends ReferenceRegion (and is often just ReferenceRegion directly).
+ *
+ * @param rdd The argument RDD
+ * @param kt The type of the argument RDD must be explicit
+ * @tparam Region The type of the argument RDD
+ */
+class RegionRDDFunctions[Region <: ReferenceRegion](rdd: RDD[Region])(implicit kt: ClassTag[Region])
```

Comment:
Using "Region" for a type parameter is confusing to me.  Perhaps change to `R` or `T`?